### PR TITLE
Allow Symfony 5 in composer.json

### PR DESCRIPTION
**Pull request by pierredup**
## Subject

Allow Symfony 5 components in composer.json

I am targeting this branch because it is backwards compatible.

## Changelog

```markdown
### Added
- Added symfony 5 components in composer.json

## Removed
- Removed support for symfony 3

```

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -21,13 +21,13 @@
...
-        "symfony/dependency-injection": "^3.4 || ^4.0",
-        "symfony/filesystem": "^3.4 || ^4.0",
-        "symfony/http-foundation": "^3.4.35 || ^4.0",
-        "symfony/http-kernel": "^3.4 || ^4.0",
+        "symfony/config": "^3.4 || ^4.0 || ^5.0",
+        "symfony/dependency-injection": "^3.4 || ^4.0 || ^5.0",
+        "symfony/filesystem": "^3.4 || ^4.0 || ^5.0",
+        "symfony/http-foundation": "^3.4.35 || ^4.0 || ^5.0",
+        "symfony/http-kernel": "^3.4 || ^4.0 || ^5.0",
         "symfony/process": "^3.4 || ^4.0",
```
Comment (pierredup) [inline composer.json]: The process component can't be upgraded to 5 without a BC break (due to the component not accepting string arguments anymore), so this should be upgraded separately for master.

File: composer.json [hunk 2]
```diff
@@ -21,13 +21,13 @@
     "require": {
         "php": "^7.1",
         "sonata-project/cache": "^2.0.0",
-        "symfony/config": "^3.4 || ^4.0",
-        "symfony/dependency-injection": "^3.4 || ^4.0",
-        "symfony/filesystem": "^3.4 || ^4.0",
-        "symfony/http-foundation": "^3.4.35 || ^4.0",
-        "symfony/http-kernel": "^3.4 || ^4.0",
+        "symfony/config": "^3.4 || ^4.0 || ^5.0",
```
Comment (core23) [inline composer.json]: Please change this to:
```suggestion
        "symfony/config": "^4.3 || ^5.0",
```

We won't support 3 major versions of symfony

File: tests/Adapter/SymfonyCacheTest.php [hunk 3]
```diff
@@ -117,7 +117,7 @@ public function testCacheAction(): void
         $this->assertSame(200, $response->getStatusCode(), 'Response should be 200');
         $this->assertSame('ok', $response->getContent(), 'Response should return "OK"');
 
-        $this->assertSame('2', $response->headers->get('Content-Length'));
+        $this->assertEquals(2, $response->headers->get('Content-Length'));
```
Comment (greg0ire) [inline tests/Adapter/SymfonyCacheTest.php]: flintCI has an issue with this, why did you change it to `assertEquals`?

Comment (pierredup) [reply tests/Adapter/SymfonyCacheTest.php]: Using `assertSame` with a string fails when using `--prefer-lowest` in composer (installing Symfony 4.3), and using an int fails with Symfony 5. So this was the easiest way to get the test to pass in both versions of Symfony

Comment (pierredup) [reply tests/Adapter/SymfonyCacheTest.php]: Seems like it might have been a bug in Symfony 4.3. From 4.3.4 onwards, this works fine with a string, but 4.3.0 - 4.3.3 works only with an int.
I have updated composer.json to set the minimum required version for http-foundation to 4.3.4

**Comment by OskarStark** [comment]
We should drop 3.4 in an separate PR first

**Comment by greg0ire** [comment]
@OskarStark why? Isn't a separate commit enough?

**Comment by OskarStark** [comment]
Fine 👍🏻

**Comment by greg0ire** [comment]
Thanks @pierredup !
